api/allocations.go

```diff
@@ -91,6 +91,7 @@ func (a *Allocations) Info(allocID string, q *QueryOptions) (*Allocation, *Query
 // long pauses on this API call.
 func (a *Allocations) Exec(ctx context.Context,
 	alloc *Allocation, task string, tty bool, command []string,
+	action string,
```

Comment:
I missed this the first time around, but fortunately it's showing up in broken tests: we can't add fields to methods in the `api` package without breaking the Go API's compatibility with potential third party consumers (like those our customers might have built).

So we're going to need a new method here. What I'd recommend is something with a signature like:

```go
func (a *Allocations) ExecWithOptions(ctx context.Context, execOpts *ExecOpts, q *QueryOptions) (exitCode int, err error) {
```

And then the `ExecOpts` struct will look like this:

```go
type ExecOpts struct {
	Alloc          *Allocation
	Task           string
	TTY            bool
	Command        []string
	Action         string
	Stdin          io.Reader
	Stdout         io.Writer
	Stderr         io.Writer
	TerminalSizeCh <-chan TerminalSize
}
```

Normally this is an unusually tedious way of writing Go, but for the public Go API it lets us add fields without breaking the signature of the method due to Go's handling of default zero values.